### PR TITLE
Fix x injector ns pod info not accurate when use default

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
The pod revision was compared wrong, which lead to inaccurate info. Actually those pods are injected and the control plane is the default:
```
istioctl x injector list
NAMESPACE                     ISTIO-REVISION POD-REVISIONS
default                       default        <non-Istio>: 8 NEEDS RESTART: 8
istio-io-tcp-traffic-shifting default        <non-Istio>: 3 NEEDS RESTART: 3
istio-operator                default        <no pods>
sample-namespace              MISSING/1-14-0 <no pods>
test                          default        <non-Istio>: 3 NEEDS RESTART: 3
```
After the revision it will be like:
```
istioctl x injector list
NAMESPACE                     ISTIO-REVISION POD-REVISIONS
default                       default        default: 8
istio-io-tcp-traffic-shifting default        default: 3
istio-operator                default        <no pods>
sample-namespace              MISSING/1-14-0 <no pods>
test                          default        default: 3
```

### CODE REVIEW

File: istioctl/cmd/injector-list.go [hunk 1]
```diff
@@ -286,8 +286,15 @@ func podCountByRevision(pods []v1.Pod, expectedRevision string) map[string]revis
 	for _, pod := range pods {
 		revision := pod.ObjectMeta.GetLabels()[label.IoIstioRev.Name]
 		revisionLabel := revision
+		// if expectedRevision is "default", which is the injector revision default value,
```
[the remark this answers is not in the document]
Comment (hanxiaop) [reply istioctl/cmd/injector-list.go]: @howardjohn Thanks for reminding. I think the bug was casused due to the wrong revision extraction. I've improved the logic.

File: istioctl/cmd/injector-list.go [hunk 2]
```diff
@@ -301,6 +303,22 @@ func podCountByRevision(pods []v1.Pod, expectedRevision string) map[string]revis
 	return retval
 }
 
+func extractRevisionFromPod(pod *v1.Pod) string {
+	rev := pod.GetLabels()[label.IoIstioRev.Name]
+	if rev != "" {
+		return rev
+	}
+	statusAnno, ok := pod.GetAnnotations()[annotation.SidecarStatus.Name]
```
Comment (howardjohn) [inline istioctl/cmd/injector-list.go:311]: why not just always use the status annotation? It is the real source of truth, not the label

Comment (hanxiaop) [reply istioctl/cmd/injector-list.go:307]: @howardjohn Fixed. Thanks

**Comment by zhlsunshine** [comment]
LGTM

**Pull request merged**